assets/js/vue/components/FundingCyclesList.vue

```diff
@@ -30,30 +39,77 @@
         data() {
             return {
+                selectedProjectDirector: null,
                 researchGroups: [],
                 researchGroupDisabled: true,
+                projectDirectorsOptions: [{ value: null, text: '[Please select an associated Project Director]' }],
+                projectDirectorIds: []
             }
         },
         methods: {
             populateResearchGroups: function(event) {
                 this.researchGroups = [];
                 this.researchGroupDisabled = true;
                 this.fundingCycles.forEach(fundingCycle => {
-                    if (fundingCycle.id == event.target.value) {
+                    if (fundingCycle.id === Number(event.target.value)) {
                         this.researchGroups = fundingCycle.researchGroups;
                         this.researchGroupDisabled = false;
                     }
                 })
             },
             selectResearchGroup: function(event) {
                 if (event.target.value) {
-                    window.open("/research-group/about/" + event.target.value, '_blank');
+                    this.openResearchGroupLandingPage(event.target.value);
                 }
+            },
+            populateProjectDirectors: function () {
+                this.fundingCycles.forEach(fundingCycle => {
+                    fundingCycle.researchGroups.forEach(researchGroup => {
+                        researchGroup.projectDirectors.forEach(projectDirector => {
+                            if (this.projectDirectorIds.indexOf(projectDirector.id) === -1) {
+                                this.projectDirectorIds.push(projectDirector.id);
+                                this.makeProjectDirectorOption(researchGroup.id, projectDirector.name);
+                            } else {
+                                this.projectDirectorIds.push(projectDirector.id);
+                                this.makeProjectDirectorOption(
+                                    researchGroup.id,
+                                    projectDirector.name + ' - ' + researchGroup.shortName
+                                );
+                            }
+                        })
+                    })
+                })
+            },
+            openResearchGroupLandingPage: function (researchGroupId) {
+                window.open("/research-group/about/" + researchGroupId, '_blank');
+            },
+            makeProjectDirectorOption: function (id, name) {
+                this.projectDirectorsOptions.push({
+                    value: id,
+                    text: name
+                })
             }
+        },
+        created() {
+            this.populateProjectDirectors();
+        },
+        watch: {
+            selectedProjectDirector: function () {
+               this.openResearchGroupLandingPage(this.selectedProjectDirector);
+            }
+        },
+        filters: {
+            truncateResearchGroupName: function (text) {
```

Comment:
Missing length parameter here.

Reply:
It is not a required parameter, by default the value is sent as the parameter in the mustache interpolations

Reply:
Yes, but what if I want to truncate each call with a different length, maybe set a default instead then.

Reply:
Yes, if you look at line 32, I've added a constant for default length